generator/ProtocolTestsGenerator/smithy-dotnet-codegen/src/main/java/software/amazon/smithy/dotnet/codegen/HttpProtocolTestGenerator.java

```diff
@@ -0,0 +1,510 @@
+            return null;
+        }
+
+        @Override
+        public Void nullNode(NullNode node) {
+            writer.writeInline("null");
+            return  null;
+        }
+
+        @Override
+        public Void numberNode(NumberNode node) {
+            if(inputShape.isTimestampShape()){
+                writer.write("ProtocolTestConstants.epoch.AddSeconds($L)", node.getValue());
+            }
+            else if(inputShape.isFloatShape()){
+                writer.writeInline("$LF",node.getValue());
+            }
+            else{
+                writer.writeInline("$L", node.getValue());
+            }
+            return null;
+        }
+
+        private Void getStructure(StructureShape structureShape, ObjectNode node){
+            // for a top level input or output, since the name of the request/response is determined by the operation i.e. operation + request
+            // we generate that name outside of the value node visitor and only generate structure members
+            if(isTopLevelInputOrOutput){
+                getStructureMembersShapes(structureShape,node);
+            }
+            // if it's not a top level input or output, then we create the symbol as we would a normal structure
+            // then generate the structure member names.
+            else{
+                var structureSymbol = context.symbolProvider().toSymbol(structureShape);
+                writer.openBlock("new $L\n{","}",structureSymbol,
+                        () -> getStructureMembersShapes(structureShape,node));
+            }
+            return  null;
+        }
+
+        private Void getStructureMembersShapes(StructureShape structureShape, ObjectNode node) {
+            node.getMembers().forEach((keyNode,valueNode) -> {
+                var memberShape = structureShape.getMember(keyNode.getValue()).orElseThrow(() ->
+                        new CodegenException("Unknown member shape: " + keyNode.getValue())
+                );
+                var targetShape = model.expectShape(memberShape.getTarget());
+
+                writer.write("$L = $C,",
+                        context.symbolProvider().toMemberName(memberShape),
+                        (Runnable) () -> valueNode.accept(new ValueNodeVisitor(targetShape,false,generatedInputOutputShapeName)));
+            });
+            return  null;
+        }
+
+        @Override
+        public Void objectNode(ObjectNode node) {
+            switch (inputShape.getType()) {
+                case STRUCTURE:
+                    getStructure((StructureShape) inputShape, node);
+                    return null;
+                case MAP:
+                    getMap((MapShape) inputShape, node);
+                    return null;
+                case DOCUMENT:
+                    getDocument((DocumentShape) inputShape, node);
+                    return null;
+                case UNION:
+                    getUnion((UnionShape) inputShape, node);
```

Comment:
better to include `return null` here too, to avoid risk or another case getting added and forgetting to return from the UNION case.